pallets/order-book/src/lib.rs

```diff
@@ -387,7 +403,7 @@ pub mod pallet {
 			asset_in: T::AssetCurrencyId,
 			asset_out: T::AssetCurrencyId,
 			buy_amount: T::Balance,
-			price: T::SellRatio,
+			max_sell_price: T::SellRatio,
```

Comment:
I feel the current `place_order()` API is enough by now, and there is no need to split. 

You choose "buy" or "sell" specifying the order on `currency_in` and `currency_out`. And the `max_sell_price` will be `1 / max_sell_price` for the cases where you want the opposite swap (as un Uniswap in this regard).

WDYT? @mustermeiszer @wischli